x-pack/plugins/apm/server/lib/anomaly_detection/get_anomaly_detection_jobs.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Logger } from 'kibana/server';
+import { PromiseReturnType } from '../../../../observability/typings/common';
+import { Setup } from '../helpers/setup_request';
+import { AnomalyDetectionJobByEnv } from '../../../typings/anomaly_detection';
+import { ML_GROUP_NAME_APM } from './create_anomaly_detection_jobs';
+
+export type AnomalyDetectionJobsAPIResponse = PromiseReturnType<
+  typeof getAnomalyDetectionJobs
+>;
+export async function getAnomalyDetectionJobs(
+  setup: Setup,
+  logger: Logger
+): Promise<AnomalyDetectionJobByEnv[]> {
+  const { ml } = setup;
+  if (!ml) {
+    return [];
+  }
+  try {
+    const mlCapabilities = await ml.mlSystem.mlCapabilities();
+    if (
+      !(
+        mlCapabilities.mlFeatureEnabledInSpace &&
+        mlCapabilities.isPlatinumOrTrialLicense
+      )
+    ) {
+      logger.warn(
+        'Anomaly detection integration is not availble for this user.'
+      );
+      return [];
+    }
+  } catch (error) {
+    logger.warn('Unable to get ML capabilities.');
+    logger.error(error);
+    return [];
+  }
+  try {
+    const { jobs } = await ml.anomalyDetectors.jobs(ML_GROUP_NAME_APM);
```

Comment:
It just occurs to me that we are retrieving jobs without specifying that we are filtering by `groups`. Will this return for any field that matches `ML_GROUP_NAME_APM`?

I'd expect something like:
```ts
const { jobs } = await ml.anomalyDetectors.jobs({group: ML_GROUP_NAME_APM });
```

or alternatively
```ts
const { jobs } = await ml.anomalyDetectors.job.byGroup(ML_GROUP_NAME_APM);
```

Similarly, if we want to retrieve jobs by job id:
```ts
const { jobs } = await ml.anomalyDetectors.job.byId('some-id');
```

Similarly, if we want to retrieve jobs by job id:
```ts
const { jobs } = await ml.anomalyDetectors.jobs({ id: 'some-id' });
```